src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -527,10 +530,17 @@ def _validate_metadata(
     python_setup: PythonSetup,
 ) -> None:
 
+    req_strings = (
+        {Requirement.parse(i) for i in requirements.req_strings}
```

Comment:
The error that comes up here is `pkg_resources.extern.packaging.requirements.InvalidRequirement`, which mypy balks at: 

```
src/python/pants/backend/python/util_rules/pex.py:19:1: error: Skipping
analyzing "pkg_resources.extern.packaging.requirements": found module but no
type hints or library stubs  [import]
```

I'll add the TODO, but it may end up taking some work to actually improve this error message.

Reply:
Yeah, we handle it like this:

https://github.com/pantsbuild/pants/blob/fd71177d18bd025e87e14322adf8a5abb8a02061/src/python/pants/backend/python/target_types.py#L807-L824

Reply:
Oh, so we're just handling the generic exception? That makes sense.